File: backend/backend/src/WebSocketComponent/Common/Api/DTOs/ClientTokenMessage.cs
using System.Text.Json.Serialization;

namespace backend.WebSocketComponent.Common.Api.DTOs;

public class ClientTokenMessage : ITypedMessage
{
    public static string TypeString => "client-token";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("token")]
    public required string ClientToken { get; set; }
}

File: frontend/src/types/token/ClientTokenMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class ClientTokenMessage implements ITypedMessage {
    public readonly type = MessageType.CLIENT_TOKEN;
    public msg: {
        token: string;
    };

    public constructor(msg: { token: string }) {
        this.msg = msg;
    }
}
